megamek/src/megamek/client/ui/dialogs/BotConfigTargetHexDialog.java

```diff
@@ -0,0 +1,170 @@
+    private final JLabel listLabel = new JLabel(Messages.getString("BotConfigDialog.hexListLabel"));
+    private final DefaultListModel<Coords> coordsListModel = new DefaultListModel<>();
+    private final TipList<Coords> coordsList = new TipList<>(coordsListModel);
+    private final ClientGUI clientGui;
+    private final IBoard board;
+
+    protected BotConfigTargetHexDialog(JFrame frame, @Nullable ClientGUI cg) {
+        super(frame, "BotConfigTargetUnitDialog", "BotConfigDialog.bcthdTitle");
+        clientGui = cg;
+        if (clientGui != null) {
+            if (clientGui.getClient().getGame().getPhase() == Phase.PHASE_LOUNGE) {
+                board = clientGui.chatlounge.getPossibleGameBoard(true);
+            } else {
+                board = clientGui.getClient().getBoard();
+            }
+        } else {
+            board = null;
+        }
+        initialize();
+        // Enter keypress
+        final KeyStroke enter = KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0);
+        getRootPane().getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(enter, OK_ACTION);
+        getRootPane().getInputMap(JComponent.WHEN_FOCUSED).put(enter, OK_ACTION);
+        getRootPane().getActionMap().put(OK_ACTION, new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent evt) {
+                okButtonActionPerformed(evt);
+            }
+        });
+        addWindowListener(new WindowAdapter() {
+            public void windowOpened(WindowEvent e) {
+                coordsField.requestFocus();
+            }
+        }); 
+        UIUtil.adjustDialog(this);
+    }
+    
+    @Override
+    protected Container createCenterPane() {
+        JPanel result = new JPanel();
+        result.setLayout(new BoxLayout(result, BoxLayout.PAGE_AXIS));
+        result.setBorder(BorderFactory.createEmptyBorder(5, 5, 5, 5));
+        
+        var coordsFieldPanel = new UIUtil.FixedYPanel();
+        coordsFieldPanel.setAlignmentX(JComponent.CENTER_ALIGNMENT);
+        coordsField.setToolTipText(UIUtil.formatSideTooltip(Messages.getString("BotConfigDialog.hexCoordsTip")));
+        coordsLabel.setLabelFor(coordsField);
+        coordsLabel.setDisplayedMnemonic(KeyEvent.VK_X);
+        coordsFieldPanel.add(coordsLabel);
+        coordsFieldPanel.add(coordsField);
+
+        var listLabelPanel = new UIUtil.FixedYPanel(); 
+        listLabel.setLabelFor(coordsList);
+        listLabel.setDisplayedMnemonic(KeyEvent.VK_S);
+        listLabelPanel.add(listLabel);
+        
+        coordsList.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+        coordsList.setLayoutOrientation(JList.VERTICAL);
+        coordsList.setCellRenderer(new BuildingHexRenderer());
+        coordsList.setFont(UIUtil.getScaledFont());
+        coordsList.setVisibleRowCount(6);
+        coordsList.setPrototypeCellValue(new Coords(-21, 22));

```

Comment:
This seems a little specific.

Reply:
If you mean the prototype cell value, that's only passed to the renderer to get the cell size when the list is empty. This prevents it from changing size when it goes from empty to containing items (the changed size doesn't appear directly, only when resizing the dialog, but then it looks a little silly)